### PR TITLE
Fix episode metadata & improve series metadata

### PR DESCRIPTION
**Pull request by Nalsai**
Fixes the episode metadata by getting it using the season and episode number. #6 
I added reading the summary from the episode xml, so that it can also be included in the metadata, and I added populating the ProductionYear for series metdata.

I also added AniDbReplaceGraves to a couple of places, where it was missing, and removing the * that is sometimes at the beginning of the series description (when it begins with "Based on").

### CODE REVIEW

**Comment by Nalsai** [comment]
After looking at metadata from other plugins, I found out that newlines need to be `<br>` for Jellyfin metadata, so I fixed that.

**Comment by Nalsai** [comment]
The show Zoku Owarimonogatari (13691) has links with urls like this `http://anidb.net/character/8962`, instead of the usual `https://anidb.net/ch8962`.
So I added support for that.

File: Jellyfin.Plugin.AniDB/Providers/AniDB/Metadata/AniDbEpisodeProvider.cs [hunk 1]
```diff
@@ -32,30 +30,38 @@ public AniDbEpisodeProvider(IServerConfigurationManager configurationManager)
...
 
-            var seriesFolder = await FindSeriesFolder(id.Value.SeriesId, cancellationToken);
-            if (string.IsNullOrEmpty(seriesFolder))
+            if (!Plugin.Instance.Configuration.IgnoreSeason && info.ParentIndexNumber > 1)
             {
                 return result;
             }
 
-            var xml = GetEpisodeXmlFile(id.Value.EpisodeNumber, id.Value.EpisodeType, seriesFolder);
+            string episodeType = "";
```
Comment (crobibero) [inline Jellyfin.Plugin.AniDB/Providers/AniDB/Metadata/AniDbEpisodeProvider.cs:57]:
```suggestion
            string episodeType = string.Empty;
```

File: Jellyfin.Plugin.AniDB/Providers/AniDB/Metadata/AniDbEpisodeProvider.cs [hunk 2]
```diff
@@ -71,71 +77,39 @@ public async Task<MetadataResult<Episode>> GetMetadata(EpisodeInfo info, Cancell
...
         public async Task<IEnumerable<RemoteSearchResult>> GetSearchResults(EpisodeInfo searchInfo, CancellationToken cancellationToken)
         {
-            var list = new List<RemoteSearchResult>();
-
-            var id = AniDbEpisodeIdentity.Parse(searchInfo.ProviderIds.GetOrDefault(ProviderNames.AniDb));
-            if (id == null)
+            if (!searchInfo.IndexNumber.HasValue || !searchInfo.ParentIndexNumber.HasValue)
             {
-                return list;
+                return new List<RemoteSearchResult>();
```
Comment (crobibero) [inline Jellyfin.Plugin.AniDB/Providers/AniDB/Metadata/AniDbEpisodeProvider.cs:87]:
```suggestion
                return Enumerable.Empty<RemoteSearchResult>();
```

File: Jellyfin.Plugin.AniDB/Providers/AniDB/Metadata/AniDbEpisodeProvider.cs [hunk 3]
```diff
@@ -71,71 +77,39 @@ public async Task<MetadataResult<Episode>> GetMetadata(EpisodeInfo info, Cancell
...
-                _configurationManager.ApplicationPaths,
-                id.Value.SeriesId,
-                cancellationToken).ConfigureAwait(false);
+            var metadataResult = await GetMetadata(searchInfo, cancellationToken).ConfigureAwait(false);
 
-            try
+            if (!metadataResult.HasMetadata)
             {
-                var metadataResult = await GetMetadata(searchInfo, cancellationToken).ConfigureAwait(false);
+                return new List<RemoteSearchResult>();
```
Comment (crobibero) [inline Jellyfin.Plugin.AniDB/Providers/AniDB/Metadata/AniDbEpisodeProvider.cs:94]:
```suggestion
                return Enumerable.Empty<RemoteSearchResult>();
```

File: Jellyfin.Plugin.AniDB/Providers/AniDB/Metadata/AniDbSeriesProvider.cs [hunk 4]
```diff
@@ -34,7 +34,7 @@ public class AniDbSeriesProvider : IRemoteMetadataProvider<Series, SeriesInfo>,
         // AniDB has very low request rate limits, a minimum of 2 seconds between requests, and an average of 4 seconds between requests
         public static readonly RateLimiter RequestLimiter = new RateLimiter(TimeSpan.FromSeconds(3), TimeSpan.FromSeconds(5), TimeSpan.FromMinutes(5));
         private static readonly int[] IgnoredTagIds = { 6, 22, 23, 60, 128, 129, 185, 216, 242, 255, 268, 269, 289 };
-        private static readonly Regex AniDbUrlRegex = new Regex(@"https?://anidb.net/\w+ \[(?<name>[^\]]*)\]");
+        private static readonly Regex AniDbUrlRegex = new Regex(@"https?://anidb.net/\w+(/\d+)? \[(?<name>[^\]]*)\]");
```
Comment (crobibero) [inline Jellyfin.Plugin.AniDB/Providers/AniDB/Metadata/AniDbSeriesProvider.cs:37]:
```suggestion
        private static readonly Regex AniDbUrlRegex = new Regex(@"https?://anidb.net/\w+(/[0-9]+)? \[(?<name>[^\]]*)\]");
```

>\d checks all Unicode digits, while [0-9] is limited to these 10 characters.

**Review by crobibero — APPROVED** [review]

**Review by h1dden-da3m0n — APPROVED** [review]

**Comment by h1dden-da3m0n** [comment]
@Nalsai just before I hit the merege here, the branch is good or you have any other changes that you want to get in?

**Comment by Nalsai** [comment]
It's good now 👍️

**Comment by h1dden-da3m0n** [comment]
Thank you for your contribution :slightly_smiling_face:

**Pull request merged**